ledger/ledger-api-auth/src/main/scala/com/digitalasset/ledger/api/auth/AuthServiceJWT.scala

```diff
@@ -66,28 +66,37 @@ class AuthServiceJWT(verifier: JwtVerifierBase) extends AuthService {
     } yield parsed
   }
 
-  private[this] def payloadToClaims(payload: AuthServiceJWTPayload): ClaimSet.Claims = {
-    val claims = ListBuffer[Claim]()
-
-    // Any valid token authorizes the user to use public services
-    claims.append(ClaimPublic)
-
-    if (payload.admin)
-      claims.append(ClaimAdmin)
-
-    payload.actAs
-      .foreach(party => claims.append(ClaimActAsParty(Ref.Party.assertFromString(party))))
-
-    payload.readAs
-      .foreach(party => claims.append(ClaimReadAsParty(Ref.Party.assertFromString(party))))
-
-    ClaimSet.Claims(
-      claims = claims.toList,
-      ledgerId = payload.ledgerId,
-      participantId = payload.participantId,
-      applicationId = payload.applicationId,
-      expiration = payload.exp,
-    )
+  private[this] val payloadToClaims: SupportedJWTPayload => ClaimSet = {
```

Comment:
Small nit: Scala has to compile a function-valued `val` differently from the equivalent "eta-expanded" `def`. Since you're only using this function fully-applied, it's slightly more efficient to define this as a method:
```suggestion
  private[this] def payloadToClaims(payload: SupportedJWTPayload): ClaimSet = payload match {
```

This compiles directly to a JVM method. For the `val` version, scalac compiles this to code that spins up a lambda "class"(*)  and calls its `apply` method, which will then invoke the method that actually has the lambda body (granted, this is only a small performance/class size hit, but I figured I'd point out this subtlety).

(*) not really a class, restrictions apply -- see `LambdaMetaFactory` and `invokedynamic`